File: ContainsDuplicate/II/ContainsDuplicateII.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        if len(nums) < 2:
            return False
        if k > len(nums):
            k = len(nums)
        s = 0
        while s+k <= len(nums):
            tmp = nums[s:s+k+1]
            tmp.sort()
            for i in range(len(tmp)-1):
                if tmp[i] == tmp[i+1]:
                    return True
            s += 1
        return False

File: ContainsDuplicate/II/ContainsDuplicateII.cpp
#include<vector>
#include<unordered_set>
using namespace std;

class Solution {
public:
	bool containsNearbyDuplicate(vector<int>& nums, int k) {
		unordered_set<int> tmp;
		if (k <= 0)
			return false;
		if (k >= nums.size())
			k = nums.size() - 1;
		for (int i = 0; i < nums.size(); i++) {
			if (i > k)
				tmp.erase(nums[i - k - 1]);
			if (tmp.find(nums[i]) != tmp.end())
				return true;
			tmp.insert(nums[i]);
		}
		return false;
	}
};